src/v2/view-builder/views/email/EnrollAuthenticatorEmailView.js

```diff
@@ -1,16 +1,79 @@
 import { loc } from 'okta';
 import BaseAuthenticatorEmailView from './BaseAuthenticatorEmailView';
+import { getCheckYourEmailTitle, getEnterCodeLink } from './AuthenticatorEmailViewUtil';
 
 const BaseAuthenticatorEmailForm = BaseAuthenticatorEmailView.prototype.Body;
 
-const Body = BaseAuthenticatorEmailForm.extend(Object.assign({
+const Body = BaseAuthenticatorEmailForm.extend(
+  Object.assign({
+    resendEmailAction: 'currentAuthenticator-resend',
 
-  resendEmailAction: 'currentAuthenticator-resend',
+    initialize() {
+      BaseAuthenticatorEmailForm.prototype.initialize.apply(this, arguments);
 
-  subtitle() {
-    return loc('oie.email.enroll.subtitle', 'login');
-  },
-}));
+      const email = this.options.appState.get('user')?.identifier || {};
+
+      const useEmailMagicLinkValue = this.isUseEmailMagicLink();
+
+      if (useEmailMagicLinkValue !== undefined) {
+        
+        this.noButtonBar = true;
+        this.events['click .enter-auth-code-instead-link'] = 'showAuthCodeEntry';
+
+        if (useEmailMagicLinkValue) {
+          this.add(getEnterCodeLink(), {
+            prepend: true,
+            selector: '.o-form-error-container',
+          });
+        } 
+  
+        this.add(getCheckYourEmailTitle(), {
+          prepend: true,
+          selector: '.o-form-error-container',
+          options: { email, useEmailMagicLinkValue },
+        });
+      } else {
+        this.subtitle = loc('oie.email.enroll.subtitle', 'login');
+      }
+    },
+
+    postRender() {
+      if (this.isUseEmailMagicLink() !== undefined) {
+        BaseAuthenticatorEmailForm.prototype.postRender.apply(this, arguments);
+        if (this.isUseEmailMagicLink()) {
+          this.showCodeEntryField(false);
+        } else {
+          this.noButtonBar = false;
+        }
+      }
+    },
+
+    isUseEmailMagicLink() {
+      return this.options.appState.get('currentAuthenticator')?.
+      contextualData?.useEmailMagicLink;
```

Comment:
nit: you may have to indent the second line here

Reply:
Did not get any failures in bacon? I think we should be good on this.